pkg/skaffold/errors/errors.go

```diff
@@ -21,14 +21,15 @@ import (
 )
 
 const (
-	Build       = phase("Build")
-	Deploy      = phase("Deploy")
-	StatusCheck = phase("StatusCheck")
-	FileSync    = phase("FileSync")
+	Build       = Phase("Build")
+	Deploy      = Phase("Deploy")
+	StatusCheck = Phase("StatusCheck")
+	FileSync    = Phase("FileSync")
+	Dev         = Phase("Dev")
 )
 
-type phase string
+type Phase string
 
-func ErrorCodeFromError(_ error, _ phase) proto.ErrorCode {
-	return proto.ErrorCode_COULD_NOT_DETERMINE
+func ErrorCodeFromError(_ error, _ Phase) proto.ErrorCode {
+	return proto.ErrorCode_ErrorCode_UNKNOWN
```

Comment:
`COULD_NOT_DETERMINE` seems more apt here?

It feels like we should have an error type that carries the ErrorCode inside, rather than returning an error-code and an error.  Then we could get rid of `STATUS_CHECK_NO_ERROR`.

Reply:
I think it would make sense to add  a new proto `ErrorDef` with those 2 fields. However, we won't be able to change that for other events `BuildEvent`, `DeployEvent` , `StatusCheckEvent` etc since IDEs are already consuming it. 

Re: `STATUS_CHECK_NO_ERR`, it was something we discussed for regarding following HTTP error codes for success.  It would also be easy for us to collect metrics if we define a success code vs "0" since the filed could be 0 is the default value for int type enum

Reply:
Looking back through this, I'm confused by the intent of this function.  It's a placeholder?  Can we hook in the pattern-matching aspects of #4045 here?

Or we we could our own `error` that carries an ErrorCode?  Like the [`os.Err*`](https://golang.org/pkg/os/#pkg-variables) variables and `os.IsExist()`.
<details>
<summary>example</summary>

```
type ErrorWithCode interface {
   error
   ErrorCode() ErrorCode
}

type errorWithCode struct {
    errorCode ErrorCode
    err error
}

func (e errorWithCode) ErrorCode() ErrorCode {
  return e.errorCode
}
func (e errorWithCode) Error() string {
  return fmt.Sprintf("%d: %s", e.errorCode, e.err.Error())
}

func NewErrorCode(errorCode ErrorCode, err error) ErrorWithCode {
    return errorWithCode{errorCode: errorCode, err: err}
}
```


</details>